common/ledger/util/leveldbhelper/leveldb_provider.go

```diff
@@ -149,6 +166,37 @@ func (h *DBHandle) Delete(key []byte, sync bool) error {
 	return h.db.Delete(constructLevelKey(h.dbName, key), sync)
 }
 
+// DeleteAll deletes all the keys that belong to the channel (dbName).
+func (h *DBHandle) DeleteAll() error {
+	iter := h.GetIterator(nil, nil)
+	defer iter.Release()
+
+	// This is common code shared by all the leveldb instances. Because each leveldb has its own key size pattern,
+	// each batch is limited by memory usage instead of number of keys. Once the batch memory usage reaches maxBatchSize,
+	// the batch will be committed.
+	numKeys := 0
+	batchSize := 0
+	batch := NewUpdateBatch()
+	for iter.Next() {
+		key := iter.Key()
+		numKeys++
+		batchSize = batchSize + len(key)
+		batch.Delete(key)
+		if batchSize >= maxBatchSize {
+			if err := h.WriteBatch(batch, true); err != nil {
+				return err
+			}
+			logger.Infof("Have removed %d entries for channel %s in leveldb %s", numKeys, h.dbName, h.db.conf.DBPath)
```

Comment:
Is the reason for having an `Info` log to show progress to the user?

Reply:
Yes.